### PR TITLE
[search] Treat "water_tap" and "services" as POIs.

### PR DESCRIPTION
**Pull request by vng**
@pastk Did I forget something?
We made a lot of classifier changes last time.

Everybody can advise here what else should we add as POI (search ranking heavily depends on it).

https://github.com/organicmaps/organicmaps/issues/5728

### CODE REVIEW

File: search/model.cpp [hunk 1]
```diff
@@ -15,24 +15,28 @@ using namespace std;
...
-                          {"highway", "raceway"},
-                          {"highway", "rest_area"},
-                          {"highway", "speed_camera"},
-                          {"natural", "beach"},
-                          {"natural", "geyser"},
-                          {"natural", "cave_entrance"},
-                          {"natural", "spring"},
-                          {"natural", "volcano"},
-                          {"waterway", "waterfall"}};
+  base::StringIL arr[] = {
```
Comment (biodranik) [inline search/model.cpp:18]: nit: const

**Review by biodranik — APPROVED** [review]

**Comment by pastk** [comment]
Types which are not in those lists won't be indexed?

Basically we should have all the types used in categories.txt there, is it right?
Here are some missing man_made features from categories.txt
```
man_made-water_well
man_made-lighthouse
man_made-tower-communication
man_made-tower
man_made-windmill
```

Also why a whole `railway` tree is in OneLevelPOIChecker?
Maybe that's the reason the search index was inflating when I tried to add railway lines into the World map? #5548

**Comment by vng** [comment]
To be indexed as a category, the feature type should:
- present in categories.txt
- have visible drawing rules or IsCategoryNondrawableType
- have name or IsPoiChecker
- do not present in exceptions like search::TypesSkipper

**Comment by vng** [comment]
- All man_made already present in the search index except a _chimney_ (even works or towers). Should we eliminate something more here?
- IsPoiChecker can increase rank for POIs results
- agree to add water_well, lighthouse, windmill
- don't agree to add towers as POIs

**Comment by vng** [comment]
Also, I'd remove regular man_made=tower, but keep man_made=tower=communication in search index.

File: indexer/ftypes_matcher.cpp [hunk 2]
```diff
@@ -498,7 +498,8 @@ AttractionsChecker::AttractionsChecker() : BaseChecker(2 /* level */)
       {"leisure", "park"},
       {"leisure", "water_park"},
       {"man_made", "lighthouse"},
-      {"man_made", "tower"},
+      {"man_made", "windmill"},
+      //{"man_made", "tower"},
```
Comment (pastk) [inline indexer/ftypes_matcher.cpp:502]: its also not clear what's the effect of the AttractionsChecker - could you add a comment please?

Comment (vng) [reply indexer/ftypes_matcher.cpp:502]: Turned out - no mean full usage for now.

File: data/categories.txt [hunk 3]
```diff
@@ -1108,7 +1108,7 @@ sw:Vipodozi
 fa:لوازم ارایشی بهداشتی
 
 shop-convenience|@food|@shop
-en:4Convenience|U+1F3EC|U+1F3EA|grocer|grocery|mini-mart|mini-market|superette
+en:4Convenience|U+1F3EC|U+1F3EA|grocery|mini-mart|mini-market|superette
```
Comment (biodranik) [inline data/categories.txt:1111]: grocer is not a typo, maybe leave it as is?

File: search/model.cpp [hunk 4]
```diff
@@ -48,7 +54,9 @@ class OneLevelPOIChecker : public ftypes::BaseChecker
   {
     Classificator const & c = classif();
 
-    auto paths = {"amenity", "historic", "office", "railway", "shop", "sport", "tourism", "craft"};
+    auto paths = {
```
Comment (biodranik) [inline search/model.cpp:57]: const?

File: search/search_integration_tests/smoke_test.cpp [hunk 5]
```diff
@@ -215,64 +218,124 @@ UNIT_CLASS_TEST(SmokeTest, CategoriesTest)
...
+      {"place"},
+      {"waterway"},
+
+      /// @todo Controversial here.
+      /// Don't have point drawing rules except text -> type will be removed for Feature with empty name.
+      {"amenity", "brothel"},
+      {"amenity", "public_bookcase"},
+      {"building", "train_station"},
+      {"leisure", "track"},
+      {"natural", "beach"},
```
Comment (biodranik) [inline search/search_integration_tests/smoke_test.cpp:267]: If I search for a beach category, will I find beaches without a name?

File: search/search_integration_tests/smoke_test.cpp [hunk 6]
```diff
@@ -215,64 +218,124 @@ UNIT_CLASS_TEST(SmokeTest, CategoriesTest)
...
-    auto id = BuildMwm(countryName, DataHeader::MapType::Country,
-                       [&](TestMwmBuilder & builder) { builder.AddSafe(poi); });
+    TestPOI withoutName({2.0, 2.0}, "", "");
+    if (IsCategoryNondrawableType(type))
+      withoutName.SetTypes({type, cafeType});
+    else
+      withoutName.SetTypes({type});
 
-    Rules rules = {ExactMatch(id, poi)};
+    auto id = BuildMwm(countryName, DataHeader::MapType::Country, [&](TestMwmBuilder & builder)
```
Comment (biodranik) [inline search/search_integration_tests/smoke_test.cpp:321]: const?

**Review by biodranik — APPROVED** [review]
Does it really close the issue referenced in the description?

File: data/categories.txt (hunk 3 above)
Comment (vng) [reply data/categories.txt:1111]: Suggest will advise correct spelling, useless imo, or should add everywhere

File: search/search_integration_tests/smoke_test.cpp (hunk 5 above)
Comment (vng) [reply search/search_integration_tests/smoke_test.cpp:267]: You will for area beaches.
This is about point styles, we don't create Features that you can't even see on map.

Comment (biodranik) [reply search/search_integration_tests/smoke_test.cpp:267]: Point beach should have a beach icon and be searchable too, no?

**Comment by vng** [comment]
Follow up here https://github.com/organicmaps/organicmaps/issues/5728#issuecomment-1685021868

I suggest to rename groceries -> grocery for the food category.

**Comment by biodranik** [comment]
The word grocery refers to the grocer’s shop or business. Grocery can also be used to describe a noun.

    Grocery bags are often used to store trash in.
    The grocery store is closed. (Note: For brevity, we can delete the word store and just use the word grocery. Thus: “The grocery store is closed.”)

The word groceries refers to the items sold at the grocery. As an aggregate noun, groceries is always plural.

    I need to pick up some groceries before I head home.
    The groceries that you bought online were delivered to your office.

I don't mind if it improves UX.

**Pull request merged**